src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -483,6 +556,61 @@ HashSet<ITypeSymbol> typeStack
 			).WithPrefixedMember( field.Name );
 		}
 
+		/// <summary>
+		/// Walks all ancestors to this type parameter to determine if it was
+		/// intended that the type be immutable.  If *any* ancestor says the
+		/// type was supposed to be immutable, then the type will be 
+		/// assumed to mandatory immutable.
+		/// </summary>
+		/// <param name="parameterIndex">The position of the parameter in the symbols TypeParameter array.</param>
+		/// <param name="symbol">The symbol containing the set of type parameters to be inspected.</param>
+		/// <returns>Returns <code>True</code> if the type parameter or any ancestor
+		/// type parameter in the specified <paramref name="parameterIndex"/> position is
+		/// marked for immutability.</returns>
+		private static bool InspectInterfaceTypeParameter(
+			int parameterIndex,
+			ITypeSymbol symbol
+		) {
+			if( symbol == default ) {
+				return false;
+			}
+
+			if( IsTypeParameterToBeImmutable( parameterIndex, symbol as INamedTypeSymbol ) ) {
+				return true;
+			}
+
+			foreach( var intf in symbol.Interfaces ) {
+				if( InspectInterfaceTypeParameter( parameterIndex, intf ) ) {

```

Comment:
This looks like it is assuming that if an interface has 2 type params, then parent interfaces must have the same type params, in the same order.  That's not true.  This compiles:
```
public interface BaseFoo<T, U> {}
public interface SubFoo<U, T> : BaseFoo<T, U> {}
public interface SubsetFoo<T> : BaseFoo<T, T> {}
public interface SupersetFoo<S, T, U, V> : BaseFoo<U, V> {}
```
The type parameters on `symbol` might be passed to parent interfaces multiple times, in a different order, or not at all.  This code does not seem to account for that.